modules/backend/formwidgets/TagList.php

```diff
@@ -110,6 +110,10 @@ public function prepareVars()
      */
     public function getSaveValue($value)
     {
+        if (is_array($value)) {
+            $value = array_filter($value);
+        }
```

Comment:
Doesn't this change now allow for the case where an empty value is passed `""` when mode is MODE_RELATION and it doesn't get filtered out in either location?

Reply:
No, this fields always returns an array in the POST data since its markup uses the array notation `[]` after the fieldName.

I should probably remove all check for `is_array()`

Reply:
@LukeTowers I've streamlined the getSaveData() method.

Reply:
These changes, along with the array naming of the singular input now means that a field that previously stored a string value will attempt to store an array value instead which will break things.

Reply:
No, the fiedls won't STORE an array, it will still store a string... it's just the Form itself that returns an array in the POST data, that didn't change.

Reply:
You can easily test all that with Winter.Test plugin under the Posts controller... this has been tested quite a bit already.

Reply:
Ah, I see now.